### PR TITLE
:art: feat : Issue-94 : Upgrade the Issue Template to use GitHub Issue forms

### PR DESCRIPTION
**Pull request by Rutam21**
# Linked Issue
It fixes https://github.com/legesher/legesher/issues/94

# Description
This PR updates the Issue templates for `Feature`, `Bug`, and `Documentation` requests to use the latest Github Issue forms.

# Methodology
Respective `yaml` files have been added that would be used to provide the newest template for any issue requests.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Rutam21** [comment]
@madiedgar Please review and suggest changes, if any. Thanks.

File: .github/ISSUE_TEMPLATE/bug.yaml [hunk 1]
```diff
@@ -0,0 +1,110 @@
+name: "🐛 Bug Report"
+description: "Submit a bug report to help us improve"
+title: "🐛 Bug Report: "
+labels: [bug]
```
Comment (madiedgar) [inline .github/ISSUE_TEMPLATE/bug.yaml:4]: Could you change the labels to reflect the same "bug" label used in the repo? Not sure what the exact syntax would have to be for the yaml file, but this should be --> `🐛  Opportunity: Bug`

File: .github/ISSUE_TEMPLATE/documentation.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+name: "📚 Documentation"
+description: "Report an issue related to documentation"
+title: "📚 Documentation: "
+labels: [documentation]
```
Comment (madiedgar) [inline .github/ISSUE_TEMPLATE/documentation.yaml:4]: Can you update this to the documentation label? ` 📖  Opportunity: Documentation`

File: .github/ISSUE_TEMPLATE/feature.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
+name: 🚀 Feature
+description: "Submit a proposal for a new feature"
+title: "🚀 Feature: "
+labels: [feature]
```
Comment (madiedgar) [inline .github/ISSUE_TEMPLATE/feature.yaml:4]: Can you change this to `🤔  Opportunity: Idea`

**Review by madiedgar — CHANGES_REQUESTED** [review]
This is a great start @rutam21 👍  Could you update the labels used to reflect the custom ones in the repos (search "Opportunity")? I'm not sure what the specific syntax would have to be for the issue templates to reference custom labels, so please look into that.

File: .github/ISSUE_TEMPLATE/bug.yaml (hunk 1 above)
Comment (Rutam21) [reply .github/ISSUE_TEMPLATE/bug.yaml]: If you have labels created that way then we can mention the same within brackets and it should work. Applying changes as suggested.

File: .github/ISSUE_TEMPLATE/documentation.yaml (hunk 2 above)
Comment (Rutam21) [reply .github/ISSUE_TEMPLATE/documentation.yaml]: Applied ✌️

File: .github/ISSUE_TEMPLATE/feature.yaml (hunk 3 above)
Comment (Rutam21) [reply .github/ISSUE_TEMPLATE/feature.yaml]: Applied ✌️

**Comment by Rutam21** [comment]
@madiedgar Please take a look and let me know if it looks good to you now.

**Review by madiedgar — APPROVED** [review]
This looks great 👍 thank you!

**Pull request merged**